### PR TITLE
SCIM v2 API to Add/Remove Users to Group

### PR DESCRIPTION
**Pull request by knikolla**
This PR introduces support for querying the `Groups` portion of the [SCIM v2 API](http://simplecloud.info) under the `/scim/v2/Groups` path.

- Listing groups will give a list of Resource Allocations present on the system.
- Adding or removing a user from a group will add or remove the user from that Resource Allocation.
- There is not yet support for creating users, that will come in a follow-up patch.

### CODE REVIEW

**Review by naved001 — APPROVED** [review]
Looks good, just left a couple of questions.

File: src/coldfront_plugin_api/tests/unit/test_groups.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        return client
+
+    def test_list_groups(self):
+        user = self.new_user()
+        project = self.new_project(pi=user)
+        allocation = self.new_allocation(project, self.resource, 1)
+
+        response = self.admin_client.get("/api/scim/v2/Groups")
+
+        self.assertEqual(response.status_code, 200)
```
[the remark this answers is not in the document]
Comment (knikolla) [reply src/coldfront_plugin_api/tests/unit/test_groups.py:36]: I'll add that, good point.

**Review by larsks — APPROVED** [review]

File: src/coldfront_plugin_api/tests/unit/test_groups.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+class TestAllocation(base.TestBase):
+
+    def setUp(self) -> None:
+        self.maxDiff = None
+        super().setUp()
+        self.resource = resource_models.Resource.objects.all().first()
+
+    @property
+    def admin_client(self):
```
[the remark this answers is not in the document]
Comment (knikolla) [reply src/coldfront_plugin_api/tests/unit/test_groups.py:24]: Done

File: src/coldfront_plugin_api/tests/unit/test_groups.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (knikolla) [reply src/coldfront_plugin_api/tests/unit/test_groups.py:48]: Done

**Review by naved001 — APPROVED** [review]

**Pull request merged**